### PR TITLE
bpo-30031: Improve queens demo

### PR DESCRIPTION
**Pull request by kapyshin**
* made it use argparse
* made it use singular form of “solutions” if needed
* added empty lines according to PEP 8

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately our records indicate you have not signed the CLA. For legal reasons we need you to sign this before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://cpython-devguide.readthedocs.io/pullrequest.html#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

**Comment by kapyshin** [comment]
Re-run the CLA check, please.

File: Tools/demo/queens.py [hunk 1]
```diff
@@ -67,19 +71,21 @@ def display(self):
...
-    if sys.argv[1:2] == ['-n']:
-        silent = 1
-        del sys.argv[1]
-    if sys.argv[1:]:
-        n = int(sys.argv[1])
-    q = Queens(n)
-    q.silent = silent
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "side_size", default=N, type=int, nargs="?",
```
Comment (louisom) [inline Tools/demo/queens.py:78]: To make string quote consistency, using `'` will better than `"`

File: Tools/demo/queens.py [hunk 2]
```diff
@@ -67,19 +71,21 @@ def display(self):
...
+        help="size of the board side")
+    parser.add_argument(
+        "-n", "--silent", default=SILENT, action="store_true",
```
Comment (lulouie) [inline Tools/demo/queens.py]: Is not intuitive to set `silent` to `-n`, I think `-s` is better than `-n`.

File: Tools/demo/queens.py (hunk 1 above)
Comment (lulouie) [inline Tools/demo/queens.py]: I'll prefer to set `-n` here, since is a `N` queen problem.

**Review by louisom — CHANGES_REQUESTED** [review]
Some small suggest, others LGTM.

**Comment by Mariatta** [comment]
Closing this PR since the issue was rejected in the bug tracker.

**Pull request closed without merging**